larray/util/misc.py

```diff
@@ -515,8 +516,28 @@ def error_handler(error, flag):
     return error_handler
 
 
+_zero_padding_pattern = re.compile('0+[1-9]+')
+
+
 def _isintstring(s):
-    return s.isdigit() or (len(s) > 1 and s[0] == '-' and s[1:].isdigit())
+    """
+    Parameters
+    ----------
+    s : str
+
+    Examples
+    --------
+    >>> _isintstring('12')
+    True
+    >>> _isintstring('-12')
+    True
+    >>> _isintstring('a1')
+    False
+    >>> _isintstring('01')
+    False
+    """
+    return (s.isdigit() or (len(s) > 1 and s[0] == '-' and s[1:].isdigit())) and \
```

Comment:
This is not very readable IMO and will not work for negative zero padded ints. What do you think of the following?
```python
def isposint(s):
    # avoid zero padded strings
    return s.isdigit() and not (len(s) > 1 and s[0] == '0')
return isposint(s) or (len(s) > 1 and s[0] == '-' and isposint(s[1:]))
```

Reply:
Much better indeed. Thanks.